### PR TITLE
feat(mojaloop/project#2553): send extra notification back to payee when a transfer is aborted

### PR DESCRIPTION
**Pull request by lewisdaly**

### CODE REVIEW

**Review by mdebarros — COMMENTED** [review]
Hey @lewisdaly, this PR is looking good.

No major issues from what I can see, but I do have a "sanity" check that you should consider.

File: src/handlers/transfers/handler.js [hunk 1]
```diff
@@ -487,9 +486,24 @@ const fulfil = async (error, messages) => {
...
+      if (action === TransferEventAction.RESERVE) {
+        // Get the updated transfer now that completedTimestamp will be different
+        // TODO: should we just modify TransferService.handlePayeeResponse to
+        // return the completed timestamp? Or is it safer to go back to the DB here?
+        const transferAborted = await TransferService.getById(transferId)
+        Logger.isInfoEnabled && Logger.info(Util.breadcrumb(location, `callbackReservedAborted--${actionLetter}1`))
+        const eventDetail = { functionality: TransferEventType.NOTIFICATION, action: TransferEventAction.RESERVED_ABORTED }
+        const reservedAbortedPayload = {
+          transferId: transferAborted.id,
+          completedTimestamp: Util.Time.getUTCString(new Date(transferAborted.completedTimestamp)),
```
[the remark this answers is not in the document]
Comment (lewisdaly) [reply src/handlers/transfers/handler.js:499]: yeah good point - I think I exercise this in the integration tests though - by doing a lookup on the database and ensuring the completed timestamp is the same... let me double check

File: src/handlers/transfers/handler.js [hunk 2]
```diff
@@ -381,22 +381,21 @@ const fulfil = async (error, messages) => {
         const eventDetail = { functionality, action }
         if (action !== TransferEventAction.RESERVE) {
           if (!isTransferError) {
-            Logger.isInfoEnabled && Logger.info(Util.breadcrumb(location, `callbackFinilized2--${actionLetter}3`))
+            Logger.isInfoEnabled && Logger.info(Util.breadcrumb(location, `callbackFinalized2--${actionLetter}3`))
```
Comment (mdebarros) [inline src/handlers/transfers/handler.js:384]: I may be blind, but I do not see a difference here? :D

File: src/shared/setup.js [hunk 3]
```diff
@@ -57,7 +57,7 @@ const migrate = (runMigrations) => {
 }
 
 const connectDatabase = async () => {
-  Logger.isDebugEnabled && Logger.debug(`Conneting to DB ${JSON.stringify(Config.DATABASE)}`)
+  Logger.isDebugEnabled && Logger.debug(`Connecting to DB ${JSON.stringify(Config.DATABASE)}`)
```
Comment (mdebarros) [inline src/shared/setup.js:60]: O_O

File: test/integration/helpers/testConsumer.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+ ******/
+'use strict'
+
+const Logger = require('@mojaloop/central-services-logger')
+const Consumer = require('@mojaloop/central-services-stream').Kafka.Consumer
+
+/**
+ * @class TestConsumer
+ * @description A Kafka consumer that listens for events from our test harness, and
+ *   makes it easy to ensure that
```
Comment (mdebarros) [inline test/integration/helpers/testConsumer.js:36]: makes it easy to ensure that....?

File: test/integration/helpers/testConsumer.js [hunk 5]
```diff
@@ -0,0 +1,163 @@
+/*****
```
Comment (mdebarros) [inline test/integration/helpers/testConsumer.js:1]: Is this necessary?

**Review by mdebarros — COMMENTED** [review]
Minor comments for review.

**Comment by mdebarros** [comment]
Closing stale PR, and re-created new PR @ https://github.com/mojaloop/central-ledger/pull/874

**Pull request closed without merging**